TICC.ino

```diff
@@ -16,17 +16,24 @@ int int_pin = 3; // physical pin
 volatile unsigned long long coarse_count = 0;
 
 // hardware connections to TDC2700 -- need to reassign these more logically (todo in Eagle)
-const int ENABLE_A = 13;
-const int INTB_A = 12;
-const int CSB_A = 11;
-const int ENABLE_B = 10;
-const int INTB_B = 9;
-const int CSB_B = 8;
-const int STOP_A = 7;
-const int STOP_B = 8;
-
-const int ChA = 0;	// channel A
-const int ChB = 1;	// channel B
+
+struct chanTDC7200_t {
+  const int ENABLE;
+  const int INTB;
+  const int CSB;
+  const int STOP;
+  int timeresult;
+  int clockresult;
+  int calresult;
+  long result;
+  long long unsigned stop_time;
+};
+
+static chanTDC7200_t ChanA = { 13, 12, 11, 6, 0, 0, 0, 0, 0 }; // I/O pins on Arduino
+static chanTDC7200_t ChanB = { 10, 9, 8, 7, 0, 0, 0, 0, 0 };
```

Comment:
> This I like! I'm too used to old-style C to have been aware that named initializers snuck in, or snuck in to Arduino. George


The arduino compiler is gcc.  Anything you can do in GCC, you should be able to do in an arduino sketch.

See https://gcc.gnu.org/onlinedocs/gcc/Designated-Inits.html

Note that you don’t have to initialize every field.  The ones you don’t specify are initialized to zero.

--
Jeremy McDermond (NH6Z)
Xenotropic Systems
mcdermj@xenotropic.com